rest/config_new.go

```diff
@@ -0,0 +1,240 @@
+
+	ServerReadTimeout  time.Duration `json:"server_read_timeout,omitempty"  help:"maximum duration.Second before timing out read of the HTTP(S) request"`
+	ServerWriteTimeout time.Duration `json:"server_write_timeout,omitempty" help:"maximum duration.Second before timing out write of the HTTP(S) response"`
+	ReadHeaderTimeout  time.Duration `json:"read_header_timeout,omitempty"  help:"The amount of time allowed to read request headers"`
+	IdleTimeout        time.Duration `json:"idle_timeout,omitempty"         help:"The maximum amount of time to wait for the next request when keep-alives are enabled"`
+
+	Pretty             bool  `json:"pretty,omitempty"               help:"Pretty-print JSON responses"`
+	MaximumConnections uint  `json:"max_connections,omitempty"      help:"Max # of incoming HTTP connections to accept"`
+	CompressResponses  *bool `json:"compress_responses,omitempty"   help:"If false, disables compression of HTTP responses"`
+	HideProductVersion bool  `json:"hide_product_version,omitempty" help:"Whether product versions removed from Server headers and REST API responses"`
+
+	TLS  TLSConfig   `json:"tls,omitempty"`
+	CORS *CORSConfig `json:"cors,omitempty"`
+}
+
+type TLSConfig struct {
+	MinimumVersion string `json:"minimum_version,omitempty"     help:"The minimum allowable TLS version for the REST APIs"`
+	CertPath       string `json:"cert_path,omitempty" help:"The TLS cert file to use for the REST APIs"`
+	KeyPath        string `json:"key_path,omitempty"  help:"The TLS key file to use for the REST APIs"`
+}
+
+type CORSConfig struct {
+	Origin      []string `json:"origin,omitempty"       help:"List of allowed origins, use ['*'] to allow access from everywhere"`
+	LoginOrigin []string `json:"login_origin,omitempty" help:"List of allowed login origins"`
+	Headers     []string `json:"headers,omitempty"      help:"List of allowed headers"`
+	MaxAge      int      `json:"max_age,omitempty"      help:"Maximum age of the CORS Options request"`
+}
+
+type LoggingConfig struct {
+	LogFilePath    string                    `json:"log_file_path,omitempty"   help:"Absolute or relative path on the filesystem to the log file directory. A relative path is from the directory that contains the Sync Gateway executable file"`
+	RedactionLevel base.RedactionLevel       `json:"redaction_level,omitempty" help:"Redaction level to apply to log output"`
+	Console        *base.ConsoleLoggerConfig `json:"console,omitempty"`
+	Error          *base.FileLoggerConfig    `json:"error,omitempty"`
+	Warn           *base.FileLoggerConfig    `json:"warn,omitempty"`
+	Info           *base.FileLoggerConfig    `json:"info,omitempty"`
+	Debug          *base.FileLoggerConfig    `json:"debug,omitempty"`
+	Trace          *base.FileLoggerConfig    `json:"trace,omitempty"`
+	Stats          *base.FileLoggerConfig    `json:"stats,omitempty"`
+}
+
+type AuthConfig struct {
```

Comment:
Everything in AuthConfig is deprecated in Lithium.  Worth discussing whether it should only be supported when using legacy config, and can be omitted from dynamic config.

Reply:
Moved the Facebook/Google auth options to a "DeprecatedConfig" struct, which cannot be set from a 3.0 config, but can still be used from the legacy config/in-memory upgrade case.